packages/network-controller/tests/NetworkController.test.ts

```diff
@@ -439,189 +276,30 @@ describe('NetworkController', () => {
                 },
               },
             ]);
-            createMetamaskProviderMock.mockReturnValue(fakeMetamaskProvider);
+            const fakeNetworkClient = buildFakeNetworkClient(fakeProvider);
+            createNetworkClientMock.mockReturnValue(fakeNetworkClient);
 
             await controller.initializeProvider();
 
-            expect(createMetamaskProviderMock).toHaveBeenCalledWith({
-              chainId: undefined,
-              engineParams: { pollingInterval: 12000 },
-              nickname: undefined,
+            expect(createNetworkClientMock).toHaveBeenCalledWith({
               rpcUrl: 'http://localhost:8545',
-              ticker: undefined,
+              chainId: toHex(1337),
```

Comment:
This is one of many instances of this kind of change. Even though the `chainId` in the provider config is a decimal number, `createNetworkClient` takes a hex string, so I had to update this across the board.

Reply:
TODO: Fix this in a separate PR